File: Algorithms/Implementaion/jumping_on_the_clouds_revisited.rb
###################

# Method to find the remaining energy after completing the game
#
# @param jump_distance [Number] distance covered by single jump
# @param clouds [Boolean Array] cloud details (Array of 0/1)
#
# @return [Number] remaining energy level
def calc_remaining_energy(
    jump_distance:,
    clouds:
)
  energy = 100
  counter = 0
  max_limit = clouds.length
  until counter >= max_limit
    energy -= jump_distance unless clouds[counter].zero?
    counter += jump_distance
    energy -= 1
  end
  energy
end
###################

jump_details = gets.strip.split(' ').map(&:to_i)
clouds = gets.strip.split(' ').map(&:to_i)

puts calc_remaining_energy(
  jump_distance: jump_details[1],
  clouds: clouds
)

File: Algorithms/Implementaion/jumping_on_the_clouds_revisited.py


def jumping_on_cloud(data, data_size, k):
    energy = 100
    start = 0
    reached = 0
    while reached == 0:
        position = start + k
        if position >= data_size:
            start = position - data_size
            position = 0
        if data[position] == 1:
            energy -= 2
        energy -= 1
        start = position
        if position == 0:
            break
    return energy


details = list(map(int, input().split()))
data = list(map(int, input().split()))

print(jumping_on_cloud(data=data, data_size=details[0], k=details[1]))
